### PR TITLE
#186 leftover: confidential voters, Name not in the List, online then in person

### LINKED ISSUE #186: Edge cases: confidential voters, Name-not-in-the-List, online-then-in-person
_Updated 29 Aug 2026 by TallyJ Builder after backlog hygiene._

**Priority:** High  
**Risk Area:** Core / Edge Cases

## Remaining
- [ ] Test confidential voter workflow (add as “Confidential X”, manual voter count bump)
- [ ] Test “Name not in the List” as a valid vote option (when enabled)
- [ ] Confirm behavior when a voter has both an online ballot and then votes in person

## Ties / extras
Tracked in #198 (not this issue).

## Already done
- Large-election performance is out of this issue’s remaining scope

**Related:** #167, #198, #194

### PR DESCRIPTION
**Pull request by glittle**
grokbot: Glen

Draft for the three remaining #186 checks. **#186 stays open.**

This implements gaps against existing v3/v4 product. It does not invent a Confidential type, a Name-not-in-the-List spoil GUID, or a new mixed-method rule.

## Product correction (56ad9bea)

A confidential voter is an ordinary person named Confidential 1 (etc.). Adding them increases calculated Eligible Voters. Checking them in counts as In Person. There is **no** override step and **no** Analyze (or other) copy that mentions confidential.

Analyze Save Values remains v3 count-override parity (Calculated / Override / Final, `manual ?? calculated`). Empty override uses calculated. That table is not part of the confidential workflow.

## What v3 already did (and v4 now matches)

**Confidential X**
- Add a normal person named Confidential 1 (or Confidential 2, …) and check them in.
- Calculated eligible and in-person go up.
- No Add Confidential button, no auto-numbering, no Eligible Voters override for this.

**Name not in the List**
- Ballot entry option (v3 optgroup), not a spoil reason GUID.
- Adding an eligible person is a valid vote; ineligible write-ins spoil.
- Election switch **Can Add People?** (`GuestTellersCanAddPeople`, v3 ExtraSetting GA, default off).
- Guests may create a person only when that switch is on. Full tellers always can.
- When it is off, guests see “(Ask head teller to add required name)”.

**Online then in person**
- Already product (#194 / Front Desk).
- Draft or Submitted online: Front Desk other method withdraws the pending row and records the method.
- Processing: refuse (`frontDesk.errors.alreadyProcessingOnline`).
- Processed: refuse (`frontDesk.errors.alreadyAcceptedOnline`).

## v4-only difference (documented)

v3 SaveManual re-ran analysis immediately. v4 Save Values persists M only. The next Analyze applies the overrides so Calculate stays behind the count-reconciliation / pending-online gate. Panel Final is always `manual ?? calculated` (not stored F).

## Tests

- `AddConfidential1_IncreasesCalculatedEligibleVoters`
- `CheckInConfidential1_CountsAsInPerson`
- `AnalyzeManualCountsTests` (general Save Values; not confidential)
- Name not in the List / online-then-in-person / guest AccessCode claims

## Out of this PR

#198 ties/extras. #168 v3 package compare. #191 two-browser UAT. #229 SignalR. #255 WhatsApp. #254 SMS.

Do not merge until review. Do not close #186.

<div><a href="https://cursor.com/agents/bc-65ca6f36-0813-5026-ab26-1aa6c6d102c2?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-65ca6f36-0813-5026-ab26-1aa6c6d102c2&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

File: backend/Helpers/GuestTellerClaims.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    public static bool IsGuestTeller(ClaimsPrincipal? user)
+    {
+        if (user == null)
+        {
+            return false;
+        }
+
+        var claim = user.FindFirst(IsTellerClaimType)?.Value;
+        return bool.TryParse(claim, out var isGuestTeller) && isGuestTeller;
```
Comment (glittle) [inline backend/Helpers/GuestTellerClaims.cs]: grokbot: **Consistency (write-gate):** New shared guest check only looks at `isTeller`. `FullTellerAccessHandler`, `TellerAccessHandler`, and `ElectionAccessHandler` all also require `authMethod == AccessCode`. Please match that here so this helper stays the single safe definition for the CreatePerson gate (today `GenerateTellerToken` sets both, but the established pattern is both claims).

File: backend/Services/TallyService.ManualCounts.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        var final = summaries.FirstOrDefault(rs => rs.ResultType == ResultTypeFinal);
+
+        var calculatedRow = calculated != null
+            ? ToCountRow(calculated)
+            : await LiveCalculatedCountsAsync(electionGuid);
+
+        var manualRow = ToCountRow(manual);
+        var finalRow = final != null
+            ? ToCountRow(final)
+            : CombineManualOverCalculated(calculatedRow, manualRow);
```
Comment (glittle) [inline backend/Services/TallyService.ManualCounts.cs]: grokbot: **Blocking:** After an election already has a stored `F` row, Save Values updates `M` but this endpoint still returns that stale `F` as `Final`. The Analyze table’s Final column then disagrees with Override until the next Calculate — so the confidential Eligible Voters override looks like it did not take effect.

v3 Final was the live Manual ?? Calculated preview (and SaveManual re-analyzed). For v4 (persist M only), this panel API should always set `Final = CombineManualOverCalculated(calculatedRow, manualRow)` so Final tracks Override immediately. Stored `F` is rewritten on the next Calculate via `CombineCalcAndManualSummaries`.

Please adjust here + cover with a test: prior `F` exists, SaveManual changes Eligible, response `Final.NumEligibleToVote` matches the new manual value (not the old `F`).

**Review by glittle — COMMENTED** [review]
grokbot: Copilot-style review of #334 (`16e86682`) for the remaining #186 edge cases.

**Blocking** (treat as must-fix before ready-for-Glen; cannot REQUEST_CHANGES on own PR)
1. `GetAnalyzeCountSummariesAsync` prefers stored `F` for Final. After Save Values updates `M` while an old `F` exists, the Analyze Final column stays stale until Calculate — Override and Final disagree, which breaks the confidential Eligible Voters workflow visibility. Always return Final as Manual ?? Calculated for this panel API (see inline).

**Also fix**
2. `GuestTellerClaims.IsGuestTeller` should require `authMethod=AccessCode` like the existing auth handlers (see inline).

**Looks good otherwise:** Guest Can Add People switch + CreatePerson gate + ballot labels/tests; SaveManual persists M without re-analyze; OpenAPI client regenerated; online-then-in-person coverage; context docs; no PII in new logs.

**Docs-only (non-blocking):** inserting the manual-counts endpoints removed the XML `<summary>` that previously sat on `SaveTieCounts`.

Keeping draft until the Final-column issue is fixed. Do not merge. Do not close #186.

**Review by glittle — COMMENTED** [review]
grokbot: Follow-up `25720c93` fixes the prior blocking items (panel Final = Manual ?? Calculated with test; GuestTellerClaims requires AccessCode; SaveTieCounts XML restored). Copilot-style re-review of #334 is clean. Marking ready for Glen — do not merge until he says yes. Do not close #186.

**Review by glittle — COMMENTED** [review]
grokbot: Follow-up on `56ad9bea` (product correction: Confidential X is an ordinary person; Analyze must not mention confidential / no Eligible override step). Copilot-style re-review of #334 is clean — prior AccessCode + panel Final fixes still hold; Save Values stays general count-override parity; Guest Can Add People / Name not in the List / online-then-in-person coverage look right. Already ready for Glen — do not merge until he says yes. Do not close #186.
